tests/framework/builder.py

```diff
@@ -4,29 +4,43 @@
 
 import json
 import os
+import tempfile
+from pathlib import Path
 from conftest import init_microvm
-from framework.artifacts import Artifact, DiskArtifact
+from framework.artifacts import Artifact, DiskArtifact, Snapshot, SnapshotType
+import framework.utils as utils
 
 
 class MicrovmBuilder:
     """Build fresh microvms or restore from snapshot."""
 
-    def __init__(self, root_path, bin_cloner_path):
+    ROOT_PREFIX = "fctest-"
+
+    def __init__(self, bin_cloner_path):
         """Initialize microvm root and cloning binary."""
-        self.root_path = root_path
         self.bin_cloner_path = bin_cloner_path
+        self.init_root_path()
+
+    def init_root_path(self):
+        """Initialize microvm root path."""
+        self.root_path = tempfile.mkdtemp(MicrovmBuilder.ROOT_PREFIX)
 
-    def build(self, kernel: Artifact, disks: [DiskArtifact], config: Artifact):
+    def build(self,
+              kernel: Artifact,
+              disks: [DiskArtifact],
```

Comment:
Tha disk configuration call would fail.

Reply:
Indeed. Can we fail fast if the disks[0] is read-only? It seems that the microVM is configured with `read-only: false` in the end, so it does not worth to reach that point if the disk is read-only.

Another topic of discussion is that we have a list as argument and I imagine that this would be the initial support for building microVMs using multiple disks. However, at this time, do we need it? If not, even if it does not hurt, can we simply change it to `DiskArtifact`?